packages/x-charts/src/context/CartesianContextProvider.tsx

```diff
@@ -224,6 +224,16 @@ function CartesianContextProvider(props: CartesianContextProviderProps) {
               ? getOrdinalColorScale({ values: axis.data, ...axis.colorMap })
               : getColorScale(axis.colorMap)),
         };
+        if (axis.scaleType === 'ordinal-time') {
```

Comment:
We could skip the craetion of the special scaletype with
```suggestion
        if (axis.data[0] instanceof Date) {
```

But that would add some kind of magic